astropy/units/tests/test_quantity.py

```diff
@@ -1055,12 +1055,31 @@ def test_quantity_pickelability():
     assert q1.unit == q2.unit
 
 
-def test_quantity_from_string():
-    with pytest.raises(TypeError):
-        q = u.Quantity(u.m * "5")
-        # the reverse should also fail once #1408 is in
+def test_quantity_initialisation_from_unit():
+    q = u.Quantity(u.m)
+    assert q.unit == u.m
+    assert q.value == 1.
+    assert u.Unit(q) == u.m
+    q = u.Quantity('m')
+    assert q.unit == u.m
+    assert q.value == 1.
+    assert u.Unit(q) == u.m
+    q = u.Quantity('')
```

Comment:
+1 : This follows from above. Nothing, i.e. `''`, is defined here. If I were to argue for this (I would not!) the assumed value should be zero. But that's ambiguous - there’s magic going on here. The value for an undefined scalar and undefined unit is `None`.